index.js

```diff
@@ -13,44 +18,45 @@ module.exports = (babel, options) => {
   } = options
   const {reactRefresh = env === 'development' && react && {}} = options
 
-  const plugins = [
-    runtime && [
-      require('@babel/plugin-transform-runtime').default,
-      {useESModules: !modules, version: require('@babel/runtime/package.json').version},
-    ],
-  ].filter(Boolean)
+  const nodeModules = {
+    include: NODE_MODULES_REGEX,
+    compact: true,
+  }
 
-  const presets = [
-    [
-      require('@babel/preset-env').default,
-      {loose, modules, targets, corejs: 3, useBuiltIns: 'entry'},
+  const nonPrecompiledPackages = {
+    exclude: PRECOMPILED_PACKAGES_REGEX,
+    plugins: [
+      runtime && [
+        require('@babel/plugin-transform-runtime').default,
+        {useESModules: !modules, version: require('@babel/runtime/package.json').version},
+      ],
+    ].filter(Boolean),
+    presets: [
+      [
+        require('@babel/preset-env').default,
+        {loose, modules, targets, corejs: 3, useBuiltIns: 'entry'},
+      ],
     ],
-  ]
+  }
 
-  const overrides = [
-    {
-      include: /node_modules/,
-      compact: true,
-    },
-    {
-      exclude: /node_modules/,
-      plugins: [
-        [require('@babel/plugin-proposal-class-properties').default, {loose}],
-        reactRefresh && [require('react-refresh/babel'), {skipEnvCheck: true, ...reactRefresh}],
-      ].filter(Boolean),
-      presets: [
-        typescript && [require('@babel/preset-typescript').default, typescript],
-        react && [
-          require('@babel/preset-react').default,
-          {development: env === 'development', useSpread: true, ...react},
-        ],
-        emotion && [
-          require('@emotion/babel-preset-css-prop').default,
-          {autoLabel: env === 'development', sourceMap: env === 'development', ...emotion},
-        ],
-      ].filter(Boolean),
-    },
-  ]
+  const nonNodeModules = {
+    exclude: NODE_MODULES_REGEX,
+    plugins: [
+      [require('@babel/plugin-proposal-class-properties').default, {loose}],
+      reactRefresh && [require('react-refresh/babel'), {skipEnvCheck: true, ...reactRefresh}],
+    ].filter(Boolean),
+    presets: [
+      typescript && [require('@babel/preset-typescript').default, typescript],
+      react && [
+        require('@babel/preset-react').default,
+        {development: env === 'development', useSpread: true, ...react},
+      ],
+      emotion && [
+        require('@emotion/babel-preset-css-prop').default,
+        {autoLabel: env === 'development', sourceMap: env === 'development', ...emotion},
+      ],
+    ].filter(Boolean),
+  }
 
-  return {plugins, presets, overrides}
+  return {overrides: [nodeModules, nonPrecompiledPackages, nonNodeModules]}
```

Comment:
Could you explain why everything now becomes part of `overrides` instead of being split into `plugins`, `presets`, and `overrides`?

Reply:
`overrides` is a way of conditionally applying plugins and presets. This diff moves what used to be in `plugins` and `presets` into a new override block with `exclude: PRECOMPILED_PACKAGES_REGEX`. In other words, rather than applying the "standard" presets/plugins unconditionally, we should only apply them if the file being transpiled is not part of a precompiled package.